server/src/main/java/org/elasticsearch/action/support/replication/TransportWriteAction.java

```diff
@@ -384,41 +384,16 @@ void run() {
         @Override
         public void failShardIfNeeded(ShardRouting replica, String message, Exception exception,
                                       Runnable onSuccess, Consumer<Exception> onPrimaryDemoted, Consumer<Exception> onIgnoredFailure) {
-
-            logger.warn((org.apache.logging.log4j.util.Supplier<?>)
```

Comment:
interested in creating a PR to remove all those ugly casts? Jason will love it.

Reply:
Sure, will do :)